document/in_all_doc_id_set_iterator_bench_test.go

```diff
@@ -0,0 +1,214 @@
+package document
+
+import (
+	"testing"
+
+	rroaring "github.com/RoaringBitmap/roaring"
+	proaring "github.com/pilosa/pilosa/roaring"
+	"github.com/willf/bitset"
+)
+
+func BenchmarkInAllDocIDSetIteratorDense(b *testing.B) {
+	benchmarkInAllDocIDSetIterator(b, testDenseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorBitsetDense(b *testing.B) {
+	benchmarkDocIDSetIteratorBitset(b, testDenseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorRRoaringBitmapDense(b *testing.B) {
+	benchmarkDocIDSetIteratorRRoaringBitmap(b, testDenseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorPilosaRoaringBitmapDense(b *testing.B) {
+	benchmarkDocIDSetIteratorPilosaRoaringBitmap(b, testDenseBenchSetup)
+}
+
+func BenchmarkInAllDocIDSetIteratorSparse(b *testing.B) {
+	benchmarkInAllDocIDSetIterator(b, testSparseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorBitsetSparse(b *testing.B) {
+	benchmarkDocIDSetIteratorBitset(b, testSparseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorRRoaringBitmapSparse(b *testing.B) {
+	benchmarkDocIDSetIteratorRRoaringBitmap(b, testSparseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorPilosaRoaringBitmapSparse(b *testing.B) {
+	benchmarkDocIDSetIteratorPilosaRoaringBitmap(b, testSparseBenchSetup)
+}
+
+func benchmarkInAllDocIDSetIterator(b *testing.B, ts testBenchSetup) {
+	bms := initBenchBitmaps(ts)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		iters := make([]DocIDSetIterator, 0, ts.numFields)
+		for _, bm := range bms {
+			iters = append(iters, newbitmapBasedDocIDIter(bm.Iterator()))
+		}
+		inAllIter := newInAllDocIDSetIterIterator(iters...)
+
+		count := 0
+		for inAllIter.Next() {
+			count++
+		}
+	}
+}
+
+func benchmarkDocIDSetIteratorBitset(b *testing.B, ts testBenchSetup) {
+	bms := initBenchBitmaps(ts)
+
+	bs1 := bitset.New(uint(ts.totalDocs))
+	bs2 := bitset.New(uint(ts.totalDocs))
+	buf := make([]uint, 256)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		bs1.ClearAll()
```

Comment:
is it fair for this benchmark to reuse the same bitset and incur the extra cost of the [ClearAll()](https://github.com/willf/bitset/blob/71fa2377963fc761fc6556dd5895dff5816d4e8c/bitset.go#L327) when there are other benchmarks that are just initializing a new bitmap/iterator?

Reply:
I imagine the bitset will be cached inside the implementation and reused, whereas there's no API support for reusing bitmaps.